language/compiler/ir_to_bytecode/src/parser.rs

```diff
@@ -16,13 +15,68 @@ use types::account_address::AccountAddress;
 // Re-export this to make it convenient for other crates.
 pub use ir_to_bytecode_syntax::ast;
 
-// Since lalrpop can't handle comments without a custom lexer, we somewhat hackily remove all the
-// comments from the input string before passing it off to lalrpop. We only support single line
-// comments for now. Will later on add in other comment types.
-fn strip_comments(string: &str) -> String {
-    // Remove line comments
-    let line_comments = Regex::new(r"(?m)//.*$").unwrap();
-    line_comments.replace_all(string, "$1").into_owned()
+/// Determine if a character is an allowed eye-visible (printable) character.
+///
+/// The only allowed printable characters are the printable ascii characters (SPACE through ~) and
+/// tabs. All other characters are invalid and we return false.
+pub fn is_permitted_printable_char(c: char) -> bool {
+    let x = c as u32;
+    ((x > 0x1F) &&  // Don't allow meta characters
```

Comment:
maybe break this out into 3 descriptive locals, my brain is having trouble reading the nested parens with comments